JetStream: cluster processing of a pull consumer creation may panic

## Defect

Server panic due to pull consumer's waitQueue being nil

#### Versions of `nats-server` and affected client libraries used:

Server v2.8.2

#### OS/Container environment:

All

#### Steps or code to reproduce the issue:

Not sure

#### Expected result:

No panic

#### Actual result:

Get this panic:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x781c5b]
goroutine 276 [running]:
github.com/nats-io/nats-server/v2/server.(*waitQueue).isEmpty(...)
#011/home/runner/work/nats-server/src/github.com/nats-io/nats-server/server/consumer.go:2413
github.com/nats-io/nats-server/v2/server.(*consumer).processWaiting(0xc00047bfd8)
#011/home/runner/work/nats-server/src/github.com/nats-io/nats-server/server/consumer.go:2770 +0x3b
github.com/nats-io/nats-server/v2/server.(*consumer).infoWithSnap(0xc000ac5c00, 0x0)
#011/home/runner/work/nats-server/src/github.com/nats-io/nats-server/server/consumer.go:2075 +0x5bb
github.com/nats-io/nats-server/v2/server.(*consumer).info(...)
#011/home/runner/work/nats-server/src/github.com/nats-io/nats-server/server/consumer.go:2021
github.com/nats-io/nats-server/v2/server.(*jetStream).processClusterCreateConsumer(0xc000338420, 0xc0016c7290, 0x0, 0x1)
#011/home/runner/work/nats-server/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:3276 +0xf65
github.com/nats-io/nats-server/v2/server.(*jetStream).processConsumerAssignment(0xc000338420, 0xc0016c7290)
#011/home/runner/work/nats-server/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:2999 +0x86f
github.com/nats-io/nats-server/v2/server.(*jetStream).applyMetaSnapshot(0xc000338420, {0xc00163c066, 0x8059741980594afc, 0x8059c65380599d36}, 0x1)
#011/home/runner/work/nats-server/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:1125 +0x8c6
github.com/nats-io/nats-server/v2/server.(*jetStream).applyMetaEntries(0xc000338420, {0xc00000eef0, 0x1, 0x0}, 0x1, 0xc000117b98)
#011/home/runner/work/nats-server/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:1383 +0x818
github.com/nats-io/nats-server/v2/server.(*jetStream).monitorCluster(0xc000338420)
#011/home/runner/work/nats-server/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:887 +0xa65
created by github.com/nats-io/nats-server/v2/server.(*Server).startGoRoutine
#011/home/runner/work/nats-server/src/github.com/nats-io/nats-server/server/server.go:3013 +0x87
```

@ripienaar reported this, but not sure if this was during a server/cluster restart or not. I have not been able to reproduce the situation where the processing of a consumer assignment detects that the consumer existed but the consumer's waitQueue is nil.

I know that this is introduced in 2.8.2 with PR #3099 because waitQueue.len() used to protect against waitQueue being nil, and I missed that. So fix is easy, but not sure how to reproduce this or if it is not a sign of a deeper problem. Will have a PR soon to fix the code.

--- a/server/consumer.go
+++ b/server/consumer.go
@@ -2410,10 +2410,13 @@ func (wq *waitQueue) isFull() bool {
 }
 
 func (wq *waitQueue) isEmpty() bool {
-	return wq.n == 0
+	return wq.len() == 0
 }
 
 func (wq *waitQueue) len() int {
+	if wq == nil {
+		return 0
+	}
 	return wq.n
 }
 

